packages/common/core-interfaces/src/disposable.ts

```diff
@@ -5,7 +5,7 @@
 
 /**
  * Base interface for objects that require lifetime management via explicit disposal.
- * @public
+ * @beta
  */
 export interface IDisposable {
```

Comment:
If there are any @alpha / /legacy uses of this it should remain @public.

Reply:
We have 3 separate IDisposable interfaces. My hope is we can align all the public code with a single one. Since the one from tree is already public and used by public types, I demoted this one which is not used by anything public. This is used by some alpha APIs, but I think not any we want to keep as part of fluid-framework/legacy.

There clearly is more work to do on our IDisposable types, and I think marking this as beta helps reflect that (and avoid public types conflicting). While this type will have to get merged with another or get promoted back to public, I don't think having it be beta for now is a problem, and should help avoid issues where third parties implement this interface expecting it to be a stable utility interface and then have issues when we want to delete or change it.

Reply:
That aspect of this change is superseded by https://github.com/microsoft/FluidFramework/pull/21184